sdk/python/feast/feature_store.py

```diff
@@ -580,12 +601,27 @@ def _group_refs(
     # view name to feature names
     views_features = defaultdict(list)
 
+    # feature set
+    feature_set = set()
+    # feature names that collide
+    feature_collision_set = set()
+
     for ref in feature_refs:
         view_name, feat_name = ref.split(":")
+        if feat_name in feature_set:
+            feature_collision_set.add(feat_name)
+        else:
+            feature_set.add(feat_name)
         if view_name not in view_index:
             raise FeatureViewNotFoundException(view_name)
         views_features[view_name].append(feat_name)
 
+    if feature_names_only and len(feature_collision_set) > 0:
+        err = ", ".join(x for x in feature_collision_set)
+        raise ValueError(
```

Comment:
Would it be possible to raise a new error type please? `FeatureNameCollisionError`? We can then test for that specific error type being raised, otherwise we might gobble up other ValueErrors.